general-concepts/ebuild-revisions/text.xml

```diff
@@ -8,37 +8,98 @@
 Ebuilds may have a Gentoo revision number associated with them. This is a
 <c>-rX</c> suffix, where <c>X</c> is an integer <d/> see <uri
 link="::ebuild-writing/file-format#File Naming Rules"/>. This
-component must only be used for Gentoo changes, not upstream releases. By
-default, <c>-r0</c> is implied.
+component must only be used for Gentoo changes, not upstream releases.
+When empty, <c>-r0</c> is implied.
 </p>
 
-<p>
-Ebuilds should have their <c>-rX</c> incremented whenever a change is made which
-will make a substantial difference to what gets installed by the package <d/> by
-substantial, we generally mean "something for which many users would want to
-upgrade". This is usually for bugfixes.
-For any such revision bump, the new ebuild should be based on the
-previous revision to ensure that fixes aren't dropped accidentally.
-</p>
+<p>Ebuild revisions usually serve two purposes:</p>
+<ol>
+  <li>
+    keeping an older version of an ebuild around when doing
+    a potentially breaking change, and
+  </li>
+
+  <li>
+    propagating rebuilds of a package when performing a meaningful
+    change that otherwise would require the users to manually rebuild
+    the package.
+  </li>
+</ol>
 
 <p>
-Simple compile fixes do <b>not</b> warrant a revision bump; this is because they do
-not affect the installed package for users who already managed to compile it.
-Small documentation fixes are also usually not grounds for a new revision.
-In particular, this applies if the package has a substantial compilation
-time; developers should use their best judgement in these circumstances.
+The developers are encouraged to use common sense when determining
+whether to introduce a new <c>-rX</c> revision. The following rule
+of thumb could be used as a guideline:
 </p>
+<ol>
+  <li>
+    If the change is likely to break the package and require users to
+    downgrade (especially if the package is stable, all non-trivial
+    changes are considered as such), then a new revision should be
+    introduced and the old one kept. If the package has stable keywords,
+    the new revision should be dropped to <c>~arch</c>
+    (see <uri link="::keywording#Keywording on Upgrades"/>).
```

Comment:
It seems as though the importance of treating stable with extra caution does not come out as strong compared to the old text. I would really like to drive home the point of "if you have the slightest doubt about a change to a stable package, revbump` here somehow.

I'm also opposed to the language of "a change is likely to break the package and require users to downgrade". In such situations, the developer should be doing more testing, not pushing it to unstable. I understand that you didn't mean it that way either. How about something like `if a developer is uncertain whether a change may break the package despite their best testing efforts and think that users are likely to benefit from having a fallback after an upgrade`